source/fem-dem/ib_particles_dem.cc

```diff
@@ -67,6 +67,35 @@ IBParticlesDEM<dim>::update_particles(
   cfd_time      = time;
 }
 
+template <int dim>
+void
+IBParticlesDEM<dim>::update_contact_candidates()
+{
+  particles_contact_candidates.resize(dem_particles.size());
+
+  double radius_factor = parameters->contact_search_radius_factor;
+
+  for (auto &particle_one : dem_particles)
+    {
+      for (auto &particle_two : dem_particles)
+        {
+          if (particle_one.particle_id != particle_two.particle_id)
+            {
+              const Point<dim> particle_one_location = particle_one.position;
+              const Point<dim> particle_two_location = particle_two.position;
+
+              if ((particle_one_location - particle_two_location).norm() <
+                  (particle_one.radius + particle_two.radius) * radius_factor)
+                {
+                  particles_contact_candidates[particle_one.particle_id].insert(
+                    particle_two.particle_id);
+                }
```

Comment:
```suggestion
          if (particle_one.particle_id < particle_two.particle_id)
            {
              const Point<dim> particle_one_location = particle_one.position;
              const Point<dim> particle_two_location = particle_two.position;

              if ((particle_one_location - particle_two_location).norm() <
                  (particle_one.radius + particle_two.radius) * radius_factor)
                {
                  particles_contact_candidates[particle_one.particle_id].insert(
                    particle_two.particle_id);
                  particles_contact_candidates[particle_two.particle_id].insert(
                    particle_one.particle_id);
                }
```

Reply:
This shouldn't be implemented as it is redundant information since only particle one will do the calculation for both particles. The original lines of codes are more appropriate and reduce memory usage. Maybe add a comment to explain that only particle 1 needs to have the particle two in its contact candidate